elearning-backend/app/Http/Controllers/API/AdminWordsController.php

```diff
@@ -9,17 +9,60 @@
 
 class AdminWordsController extends Controller
 {
+    // Parse choices
+    private function getChoices($choices)
+    {
+        return json_decode($choices);
+    }
+
     // Fetch words from selected category
-    public function index($id)
+    public function index($id = null)
     {
-        $words = Category::where('id', $id)->quizzes()->get();
+        $words = "";
+        $data = [];
+
+        if ($id === null)
+        {
+            $words = Word::with('category')->orderBy('category_id')->get();
+        }
+        else {
+            $words = Word::where("category_id", $id)->with('category')->orderBy('category_id')->get();
+        }
+
+        foreach ($words as $word) {
+            $choices = json_decode($word->choices);
+            $wordArray = [
+                "category" => $word->category,
+                "id" => $word->id,
+                "category_id" => $word->category_id,
+                "word" => $word->word,
+                "correct_answer" => $word->correct_answer,
+                "choices" => $choices
+            ];
+
+            array_push($data, $wordArray);
+        }
```

Comment:
You can use laravel map collection instead of foreach. no need to array push. you can build an object/collection inside map

```
$words->map(function ($word){
$word->choices = json_decode($word->choices);
$word= [
...
]
return $word;
})
```